### PR TITLE
[azure-cli] - add support for noble numbat

### PR DESCRIPTION
**Pull request by gauravsaini04**
**Feature name**:
 
 * Azure-Cli
 
 **Description**:
 
 This PR introduces the following functionality :
 
 * Upon introduction of `Ubuntu-24.04 (noble numbat)`, thus adding support for `noble (numbat)`
 
 _Changelog_:
 
 * Updated `install.sh`
 * Updated tests to validate the above
 
 **Checklist**:
 
 * [x]   Checked that applied changes work as expected

### CODE REVIEW

File: src/azure-cli/install.sh [hunk 1]
```diff
@@ -18,7 +18,7 @@ AZ_INSTALLBICEP=${INSTALLBICEP:-false}
 INSTALL_USING_PYTHON=${INSTALLUSINGPYTHON:-false}
 MICROSOFT_GPG_KEYS_URI="https://packages.microsoft.com/keys/microsoft.asc"
 AZCLI_ARCHIVE_ARCHITECTURES="amd64 arm64"
-AZCLI_ARCHIVE_VERSION_CODENAMES="stretch bookworm buster bullseye bionic focal jammy"
+AZCLI_ARCHIVE_VERSION_CODENAMES="stretch bookworm buster bullseye bionic focal jammy noble"
```
Comment (samruddhikhandale) [inline src/azure-cli/install.sh:21]: Can we update minor version for this Feature?

File: test/azure-cli/scenarios.json [hunk 2]
```diff
@@ -9,6 +9,16 @@
       }
     }
   },
+  "install_bicep_noble": {
```
Comment (samruddhikhandale) [inline test/azure-cli/scenarios.json:12]: Can we remove this test? We already have workflows in place. Thanks!

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]

File: src/azure-cli/install.sh (hunk 1 above)
Comment (gauravsaini04) [reply src/azure-cli/install.sh:21]: done

File: test/azure-cli/scenarios.json (hunk 2 above)
Comment (gauravsaini04) [reply test/azure-cli/scenarios.json]: removed

**Review by samruddhikhandale — APPROVED** [review]

**Pull request merged**